ZFSin/zfs/module/zfs/zfs_vnops_windows.c

```diff
@@ -4774,6 +4775,23 @@ dispatcher(
 	}
 #endif
 	validity_check = *((uint64_t *)Irp);
+	IrpSp = IoGetCurrentIrpStackLocation(Irp);
+
+	dprintf("%s: enter: major %d: minor %d: %s: type 0x%x\n", __func__, IrpSp->MajorFunction, IrpSp->MinorFunction,
+		major2str(IrpSp->MajorFunction, IrpSp->MinorFunction), Irp->Type);
+
+	zmo = DeviceObject->DeviceExtension;
+
+	extern PDRIVER_DISPATCH STOR_MajorFunction[IRP_MJ_MAXIMUM_FUNCTION + 1];
+
+	if (DeviceObject != ioctlDeviceObject) {
+		if (zmo == NULL || ((zmo->type != MOUNT_TYPE_DCB) && (zmo->type != MOUNT_TYPE_VCB))) {
+			if (STOR_MajorFunction[IrpSp->MajorFunction] != NULL) {
+				//dprintf("Relaying IRP to STORport\n");
+				return STOR_MajorFunction[IrpSp->MajorFunction](DeviceObject, Irp);
+			}
```

Comment:
if NULL then no reason to continue below in the routine. we should jump at the end of the routine and move this code here:
// Got a request we don't care about?
			Status = STATUS_INVALID_DEVICE_REQUEST;				Status = STATUS_INVALID_DEVICE_REQUEST;
			Irp->IoStatus.Information = 0;				Irp->IoStatus.Information = 0;

Reply:
OK. I will do that.